beacon-chain/cache/committee.go

```diff
@@ -108,10 +109,9 @@ func (c *CommitteeCache) AddCommitteeShuffledList(committees *Committees) error
 	c.lock.Lock()
 	defer c.lock.Unlock()
 
-	if err := c.CommitteeCache.AddIfNotPresent(committees); err != nil {
+	if err := c.CommitteeCache.Add(committees); err != nil {
```

Comment:
Changing this to `Add` so `ProposerIndices` can get updated.
Rationale: https://github.com/prysmaticlabs/prysm/pull/4528/files#diff-ca47ed260f241c5e0485afee533e7a36R430

Reply:
I don't understand why this is needed? Do we trigger update of ProposerIndices on Add?

Reply:
Yeah we do. See: https://github.com/prysmaticlabs/prysm/pull/4528/files#diff-410eda7f72e0233b613ba03a207e91abR172. That's why this needs to be `Add` because proposer indices don't get precomputed with one epoch look ahead. So every start of the epoch, we need to add proposer indices in. Took me a while to figure out why isn't proposer indices updated for run time

Reply:
They can't be precomputed too?

Reply:
Yeah. They are precomputed. Just not with one epoch look ahead. Everything else (ie shuffled indices, sorted indices) are precomputed with one epoch look ahead.

At epoch 10: 
update epoch 10's object `&Cache{ShuffledIndices, SortedIndices, ProposerIndices}`
update epoch 11's object `&Cache{ShuffledIndices, SortedIndices}`

At epoch 11:
update epoch 11's object `&Cache{ShuffledIndices, SortedIndices, ProposerIndices}`
update epoch 12's object `&Cache{ShuffledIndices, SortedIndices}`

`ProposerIndices` needs to be added at current epoch